File: pranamitra-backend/src/main/java/com/pranamitra/service/NotificationService.java
package com.pranamitra.service;

import java.util.List;
import com.pranamitra.entity.Notification;
import com.pranamitra.entity.User;
import com.pranamitra.enums.RoleType;

public interface NotificationService {

    void createNotification(User recipient, String title, String message, String type);

    void createNotificationForRole(RoleType roleType, String title, String message, String type);

    List<Notification> getNotificationsForUser(Long userId);

    long getUnreadCount(Long userId);

    void markAsRead(Long notificationId);

    void markAllAsRead(Long userId);

    void deleteNotification(Long notificationId);
}

File: pranamitra-frontend copy/src/services/notificationService.js
import api from "../api/axios";

const notificationService = {
  getNotifications: async (userId) => {
    const response = await api.get(`/notifications/user/${userId}`);
    return response.data.data; // List of Notification objects
  },

  getUnreadCount: async (userId) => {
    const response = await api.get(`/notifications/user/${userId}/unread-count`);
    return response.data.data; // Long count value
  },

  markAsRead: async (id) => {
    const response = await api.put(`/notifications/${id}/read`);
    return response.data;
  },

  markAllAsRead: async (userId) => {
    const response = await api.put(`/notifications/user/${userId}/read-all`);
    return response.data;
  },

  deleteNotification: async (id) => {
    const response = await api.delete(`/notifications/${id}`);
    return response.data;
  },
};

export default notificationService;
